### PR TITLE
DROOLS-2848 : Make the Verifier Web Worker connector reusable for different V&V types

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.jboss.org/browse/DROOLS-2848

Some changes I found necessary when adding the V&V for DMN tables. I removed the changes for the table. Just pushing these in so the code does not get stale.

The PR bundle:
https://github.com/kiegroup/kie-wb-common/pull/2295
https://github.com/kiegroup/drools-wb/pull/1004

### CODE REVIEW

**Comment by Rikkola** [comment]
@jomarko @manstis If you could take a look. I'm just moving code from drools-wb and added some shared base types that can be used by any Verifier "adapter".

**Comment by Rikkola** [comment]
And I used the old ticket since this work should have been included in that task.

File: kie-wb-common-services/kie-wb-common-verifier/kie-wb-common-verifier-reporting-client/src/main/java/org/kie/workbench/common/services/verifier/reporting/client/analysis/Poster.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+        final String json = MarshallingWrapper.toJSON(object);
+
+        LOGGER.finest("Sending: " + json);
+
+        worker.postMessage(json);
+    }
+
+    public void post(Initialize initialize) {
+        postObject(initialize);
```
Comment (manstis) [inline kie-wb-common-services/kie-wb-common-verifier/kie-wb-common-verifier-reporting-client/src/main/java/org/kie/workbench/common/services/verifier/reporting/client/analysis/Poster.java]: Now.... you could make the parameter `final` but that's not my main point.. `Initialize` is an _empty_ class so posting it doesn't appear to do much? Unless there are sub-classes of `Initialize` somewhere (I see `DrlInitialize`) in which case you could make it abstract?

File: kie-wb-common-services/kie-wb-common-verifier/kie-wb-common-verifier-reporting-client/src/main/java/org/kie/workbench/common/services/verifier/reporting/client/analysis/Poster.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    public void post(RequestStatus requestStatus) {
```
Comment (manstis) [inline kie-wb-common-services/kie-wb-common-verifier/kie-wb-common-verifier-reporting-client/src/main/java/org/kie/workbench/common/services/verifier/reporting/client/analysis/Poster.java]: You could make it `final`

File: kie-wb-common-services/kie-wb-common-verifier/kie-wb-common-verifier-reporting-client/src/main/java/org/kie/workbench/common/services/verifier/reporting/client/analysis/Poster.java (hunk 1 above)
Comment (Rikkola) [reply kie-wb-common-services/kie-wb-common-verifier/kie-wb-common-verifier-reporting-client/src/main/java/org/kie/workbench/common/services/verifier/reporting/client/analysis/Poster.java]: Good point, should make an interface or abstract class out of it.

**Comment by Rikkola** [comment]
Rebased and updated, but still I'll check the dependency notes @jomarko brought up.

File: kie-wb-common-services/kie-wb-common-verifier/kie-wb-common-verifier-reporting-client/pom.xml [hunk 3]
```diff
@@ -85,6 +85,17 @@
       <artifactId>gwtbootstrap3</artifactId>
     </dependency>
 
+    <dependency>
+      <groupId>org.jboss.errai</groupId>
+      <artifactId>errai-html5</artifactId>
```
[the remark this answers is not in the document]
Comment (Rikkola) [reply kie-wb-common-services/kie-wb-common-verifier/kie-wb-common-verifier-reporting-client/pom.xml]: com.google.gwt.webworker.client.Worker that is used to create the webworker is actually using this. Not sure why the package name is not Errai related.

**Comment by Rikkola** [comment]
jenkins please retest this

**Comment by Rikkola** [comment]
jenkins please retest this

**Comment by Rikkola** [comment]
jenkins please retest this

**Comment by Rikkola** [comment]
@manstis Both green. Could you merge please. Thank you.

**Pull request merged**